File: frontend-react/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  description: string;
  price: number;
  category: string;
  brand: string;
  image: string;
  images?: string[];
  attributes: ProductAttribute[];
  inStock: boolean;
  rating?: number;
  reviews?: number;
  createdAt: string;
  updatedAt: string;
  active?: boolean;
  featured?: boolean;
  featureStart?: string;
  featureEnd?: string;
  availableQuantity?: number;
}

export interface ProductAttribute {
  name: string;
  value: string;
  type: 'size' | 'color' | 'material' | 'other';
}

export interface ProductFilter {
  categories: string[];
  priceRange: {
    min: number;
    max: number;
  };
  brands: string[];
  attributes: {
    [key: string]: string[];
  };
  inStockOnly: boolean;
}

export interface ProductSort {
  field: 'name' | 'price' | 'rating' | 'createdAt';
  direction: 'asc' | 'desc';
}

export interface PaginationInfo {
  page: number;
  pageSize: number;
  total: number;
  totalPages: number;
}

export interface SearchResult {
  products: Product[];
  pagination: PaginationInfo;
  filters: ProductFilter;
  sort: ProductSort;
}

File: backend/inventory-service/src/main/java/com/example/inventoryservice/dto/Product.java
package com.example.inventoryservice.dto;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.Map;

@Data
public class Product {

    private String id;
    private String name;
    private String description;
    private String category;
    private Double price;
    private Boolean active = true;
    private Boolean featured = false;
    private LocalDateTime featureStart = LocalDateTime.now();
    private LocalDateTime featureEnd = LocalDateTime.now();
    private Integer availableQuantity = 0;
    private Boolean inStock = false;
    private Map<String, Object> attributes = new HashMap<>();
    private String image;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}